src/languages/clickhouse/clickhouse.formatter.ts

```diff
@@ -0,0 +1,362 @@
+  'REFRESH {EVERY | AFTER}',
+  'RANDOMIZE FOR',
+  'APPEND',
+  'APPEND TO',
+  // https://clickhouse.com/docs/sql-reference/statements/delete
+  'DELETE FROM',
+  // https://clickhouse.com/docs/sql-reference/statements/explain
+  'EXPLAIN [AST | SYNTAX | QUERY TREE | PLAN | PIPELINE | ESTIMATE | TABLE OVERRIDE]',
+  // https://clickhouse.com/docs/sql-reference/statements/grant
+  'GRANT ON CLUSTER',
+  'GRANT CURRENT GRANTS',
+  'WITH GRANT OPTION',
+  // https://clickhouse.com/docs/sql-reference/statements/revoke
+  'REVOKE ON CLUSTER',
+  'ADMIN OPTION FOR',
+  // https://clickhouse.com/docs/sql-reference/statements/check-table
+  'CHECK TABLE',
+  'PARTITION ID',
+  // https://clickhouse.com/docs/sql-reference/statements/describe-table
+  '{DESC | DESCRIBE} TABLE',
+]);
+
+const reservedSetOperations = expandPhrases([
+  // https://clickhouse.com/docs/sql-reference/statements/select/union
+  'UNION [ALL | DISTINCT]',
+  // https://clickhouse.com/docs/sql-reference/statements/parallel_with
+  'PARALLEL WITH',
+]);
+
+const reservedJoins = expandPhrases([
+  // https://clickhouse.com/docs/sql-reference/statements/select/join
+  '[GLOBAL] [INNER|LEFT|RIGHT|FULL|CROSS] [OUTER|SEMI|ANTI|ANY|ALL|ASOF] JOIN',
+]);
+
+const reservedKeywordPhrases = expandPhrases([
+  '{ROWS | RANGE} BETWEEN',
+  'ALTER MATERIALIZE STATISTICS',
+  'RENAME TO',
+]);
+
+// https://clickhouse.com/docs/sql-reference/syntax
+export const clickhouse: DialectOptions = {
+  name: 'clickhouse',
+  tokenizerOptions: {
+    reservedSelect,
+    reservedClauses: [...reservedClauses, ...standardOnelineClauses, ...tabularOnelineClauses],
+    reservedSetOperations,
+    reservedJoins,
+    reservedKeywordPhrases,
+
+    reservedKeywords: keywords,
+    reservedDataTypes: dataTypes,
+    reservedFunctionNames: functions,
+    extraParens: ['[]'],
+    lineCommentTypes: ['#', '--'],
+    nestedBlockComments: false,
+    underscoresInNumbers: true,
+    stringTypes: ['$$', "''-qq-bs"],
+    identTypes: ['""-qq-bs', '``'],
+    paramTypes: {
+      // https://clickhouse.com/docs/sql-reference/syntax#defining-and-using-query-parameters
+      custom: [
+        {
+          regex: String.raw`\{\s*[^:]+:[^}]+\}`,
+          key: v => {
+            const match = /\{([^:]+):/.exec(v);
+            return match ? match[1].trim() : v;
+          },
+        },
+      ],
+    },
+    operators: [
+      // Arithmetic
+      '%', // modulo
+
+      // Ternary
+      '?',
+      ':',
+
+      // Lambda creation
+      '->',
+    ],
+    postProcess,
+  },
+  formatOptions: {
+    onelineClauses: [...standardOnelineClauses, ...tabularOnelineClauses],
+    tabularOnelineClauses,
+  },
+};
+
+/**
+ * Converts IN and ANY from RESERVED_FUNCTION_NAME to RESERVED_KEYWORD
+ * when they are used as operators (not function calls).
+ *
+ * IN operator: foo IN (1, 2, 3) - IN comes after an identifier/expression
+ * IN function: IN(foo, 1, 2, 3) - IN comes at start or after operators/keywords
```

Comment:
I see `in` listed in section titled [Functions for Implementing the IN Operator](https://clickhouse.com/docs/sql-reference/functions/in-functions)

This gives me the impression that `in` being available as a function is more of an implementation detail of Clickhouse and one normally wouldn't use it like so.